spec/integ/megolm-integ.spec.ts

```diff
@@ -672,118 +672,116 @@ describe("megolm", () => {
+        await aliceTestClient.start();
+        // an encrypted room with just alice
+        const syncResponse = {
+            next_batch: 1,
+            rooms: {
+                join: {},
+            },
+        };
+        syncResponse.rooms.join[ROOM_ID] = {
+            state: {
+                events: [
+                    testUtils.mkEvent({
+                        type: 'm.room.encryption',
+                        skey: '',
+                        content: {
+                            algorithm: 'm.megolm.v1.aes-sha2',
+                        },
+                    }),
+                    testUtils.mkMembership({
+                        mship: 'join',
+                        sender: aliceTestClient.userId,
+                    }),
+                ],
+            },
+        };
+        aliceTestClient.httpBackend.when('GET', '/sync').respond(200, syncResponse);
 
-        let p2pSession;
-        let inboundGroupSession;
-        let decrypted;
+        // the completion of the first initialsync hould make Alice
+        // invalidate the device cache for all members in e2e rooms (ie,
+        // herself), and do a key query.
+        aliceTestClient.expectKeyQuery(
+            getTestKeysQueryResponse(aliceTestClient.userId),
+        );
 
-        return aliceTestClient.start().then(function() {
-            // an encrypted room with just alice
-            const syncResponse = {
-                next_batch: 1,
-                rooms: {
-                    join: {},
-                },
-            };
-            syncResponse.rooms.join[ROOM_ID] = {
-                state: {
-                    events: [
-                        testUtils.mkEvent({
-                            type: 'm.room.encryption',
-                            skey: '',
-                            content: {
-                                algorithm: 'm.megolm.v1.aes-sha2',
-                            },
-                        }),
-                        testUtils.mkMembership({
-                            mship: 'join',
-                            sender: aliceTestClient.userId,
-                        }),
-                    ],
-                },
-            };
-            aliceTestClient.httpBackend.when('GET', '/sync').respond(200, syncResponse);
+        await aliceTestClient.httpBackend.flushAllExpected();
 
-            // the completion of the first initialsync hould make Alice
-            // invalidate the device cache for all members in e2e rooms (ie,
-            // herself), and do a key query.
-            aliceTestClient.expectKeyQuery(
-                getTestKeysQueryResponse(aliceTestClient.userId),
-            );
+        // start out with the device unknown - the send should be rejected.
+        try {
+            await aliceTestClient.client.sendTextMessage(ROOM_ID, 'test');
+            throw new Error("sendTextMessage succeeded on an unknown device");
```

Comment:
Rather than putting a `throw` inside the `try` block like this, I think it would be clearer to use Jest's `expect(...).rejects.toThrow(...)` condition

Reply:
Hmmm, it looks like I can use Jest conditions to check that it throws the right kind of error, but I'm not seeing a way to inspect the error's properties in more detail like we're doing here; do you know a way to do that in Jest?

Reply:
I think you should be able to do something like

```typescript
.toThrow({
  name: "UnknownDeviceError",
  devices: { [aliceTestClient.userId]: { DEVICE_ID: expect.anything() } },
})
```

Reply:
Nope, looks like Jest will only let you test the error message, or test that the error object is an instance of a specific class, but nothing more detailed than that :/ https://jestjs.io/docs/expect#tothrowerror